packages/ibm-products/src/components/SidePanel/SidePanel.tsx

```diff
@@ -279,21 +278,91 @@ export let SidePanel = React.forwardRef(
 
     const shouldReduceMotion = useReducedMotion();
 
+    // Title animation on scroll related state
+    const [labelTextHeight, setLabelTextHeight] = useState<any>(0);
+    const [prevScrollTop, setPrevScrollTop] = useState<number | undefined>(0);
+
     useEffect(() => {
-      setDoAnimateTitle(animateTitle);
-    }, [animateTitle]);
+      if (open && !titleRef?.current) {
+        setDoAnimateTitle(false);
+      } else {
+        setDoAnimateTitle(animateTitle);
+      }
+    }, [animateTitle, open]);
+
+    const setTitleItemsStyleProps = useCallback(
```

Comment:
Can you rename this method name as this naming convention usually used for setState ?